src/EnergyPlus/DaylightingManager.cc

```diff
@@ -5904,15 +5905,16 @@ void DayltgHitObstruction(EnergyPlusData &state,
 
         // Lambda function for the octree to test for surface hit and update transmittance if hit
         auto solarTransmittance = [=, &state, &R1, &RN, &hit, &ObTrans](SurfaceData const &surface) -> bool {
-            if (!surface.ShadowSurfPossibleObstruction) return false; // Do Consider separate octree without filtered surfaces
+            int const ObsSurfNum = surface.Index;
+            if (!state.dataSurface->SurfShadowPossibleObstruction(ObsSurfNum)) return false; // Do Consider separate octree without filtered surfaces
```

Comment:
Since a single instance of `Surface` is a parameter to this lambda function, is it possible that the lookup of `state.dataSurface->SurfShadowPossibleObstruction(ObsSurfNum)` is more work than fetching `surface.ShadowSurfPossibleObstruction`?

Reply:
It is. However, given that this function was written when all data fields were part of the surface struct, it is also possible that this `lambda` needs to be written to take a `SurfIdx` argument rather than a `Surface &` argument. Either way, I don't think this is a huge immediate problem.

Reply:
The reason I asked about this is because the performance test for 15zonePSZ shows about 38 million more "cost" on this branch than develop (0.24% more). About 10m is coming from daylighting functions. Small overall, but just curious why these code changes would result in any increase at all.

Reply:
Yes. I found this piece suspicious when I rewrote this part to fit the new array structure into the lambda function. I actually had to add an `Index` to `Surface` to find the SurfNum. (Potentially `Index` should be useful as there are still places we find Surface Index by `Name` using `FindItemInList`)
I didn't realize it would create such performance reduction. I actually thought about rewriting the lambda function, but the syntax at this part looked quite confusing to me. Since it's potentially a performance bottleneck I'll take another look.

Reply:
I see. This one change shouldn't be that much, unless daylighting functions are called much more frequently than they need to be. Overall, the increase in instruction count probably has to do with the fact that we have to call many more array constructors & initializers. Some of that will go away if we initialize explicitly using one or several large loops as opposed to using `dimension`. Definitely things to keep markers on.

Reply:
A few things. 

1. It's fine to have both a `SurfShadowPossibleObstruction` array and a `SurfShadowPossibleObstruction` field. If they don't change over the life of the model, then there is no overhead associated with keeping them in sync.
2. It's also probably fine to leave `SurfShadowPossibleObstruction` as a field since I don't imagine it is used in the kinds of loops that would be amenable to vectorization anyway.
3. You could change the `PierceSurface`/`SurfaceOctree` functions to take `SurfIndex` instead of `Surface &` as an argument. If we are worried about the performance of those functions there are possibly other things we can do to make them faster, not sure if these mods belong in this branch/PR.